viewer-prototype/src/node/trace-server-service.ts

```diff
@@ -4,8 +4,13 @@ import { TraceServerConfigService } from '../common/trace-server-config';
 @injectable()
 export class TraceServerServiceImpl implements TraceServerConfigService {
 
-    async startTraceServer(path: string | undefined, port: number | undefined): Promise<void> {
-        spawn(`${path}`, ['-vmargs', `-Dtraceserver.port=${port}`]);
+    startTraceServer(path: string | undefined, port: number | undefined): Promise<true | string> {
+        return new Promise((resolve) => {
+            const server = spawn(`${path}`, ['-vmargs', `-Dtraceserver.port=${port}`]);
+            server.on('error', err => resolve(err.message));
+            server.on('message', () => resolve(true));
+        });
+
     }
 
     async stopTraceServer(port: number | undefined): Promise<void> {
```

Comment:
Can we also capture errors stopping the server?

Reply:
Done